cli/src/main/java/com/box/l10n/mojito/cli/command/BranchRemoveCommand.java

```diff
@@ -0,0 +1,67 @@
+package com.box.l10n.mojito.cli.command;
+
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.box.l10n.mojito.cli.ConsoleWriter;
+import com.box.l10n.mojito.cli.command.param.Param;
+import com.box.l10n.mojito.rest.client.AssetClient;
+import com.box.l10n.mojito.rest.client.RepositoryClient;
+import com.box.l10n.mojito.rest.entity.Branch;
+import com.box.l10n.mojito.rest.entity.Repository;
+import org.fusesource.jansi.Ansi;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Component
+@Scope("prototype")
+@Parameters(commandNames = {"branch-remove", "br"}, commandDescription = "remove branch")
+public class BranchRemoveCommand extends Command {
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(BranchRemoveCommand.class);
+
+    @Autowired
+    ConsoleWriter consoleWriter;
+
+    @Autowired
+    RepositoryClient repositoryClient;
+
+    @Autowired
+    AssetClient assetClient;
+
+    @Autowired
+    CommandHelper commandHelper;
+
+    @Parameter(names = {Param.REPOSITORY_LONG, Param.REPOSITORY_SHORT}, arity = 1, required = true, description = Param.REPOSITORY_DESCRIPTION)
+    String repositoryParam;
+
+    @Parameter(names = {"-b", "--branch"}, arity = 1, required = true, description = "branch")
+    String branchName;
+
+
+    @Override
+    public void execute() throws CommandException {
+        consoleWriter.newLine().a(String.format("Marking branch deleted in repository %s for ", repositoryParam)).fg(Ansi.Color.CYAN).a(repositoryParam).println(2);
+        Repository repository = commandHelper.findRepositoryByName(repositoryParam);
+        Branch branchToRemove = repositoryClient.getBranch(repository.getId(), branchName);
+
+        if (branchToRemove == null) {
+            throw new CommandException(String.format("Cannot find branch in %s by branchName %s.", repositoryParam, branchName));
+        }
+
+        Set<Long> assetIds = new HashSet<>();
+        assetIds.addAll(assetClient.getAssetIds(repository.getId(), false, false, branchToRemove.getId()));
+        consoleWriter.newLine().a(String.format("AssetIds are fetched for %s : ", branchName)).fg(Ansi.Color.CYAN).a(assetIds.toString()).println(2);
+        assetClient.deleteAssetsInBranch(assetIds, branchToRemove.getId());
```

Comment:
this marks the assets as deleted but the branch itself won't be marked as deleted as far as I remember and that's what is looked at to decide if it should be merged or not. We need this CLI to change Branch#deleted as well

Reply:
mark branch as deleted when calling remove branch CLI in addition to remove assets in branch.